src/main/java/com/capitalone/dashboard/collector/WhiteSourceCollectorTask.java

```diff
@@ -0,0 +1,187 @@
+    public WhiteSourceCollectorTask(TaskScheduler taskScheduler,
+                                    WhiteSourceCollectorRepository whiteSourceCollectorRepository,
+                                    WhiteSourceComponentRepository whiteSourceComponentRepository,
+                                    LibraryPolicyResultsRepository libraryPolicyResultsRepository,
+                                    WhiteSourceClient whiteSourceClient,
+                                    WhiteSourceSettings whiteSourceSettings) {
+        super(taskScheduler, "WhiteSource");
+        this.whiteSourceCollectorRepository = whiteSourceCollectorRepository;
+        this.whiteSourceComponentRepository = whiteSourceComponentRepository;
+        this.libraryPolicyResultsRepository = libraryPolicyResultsRepository;
+        this.whiteSourceClient = whiteSourceClient;
+        this.whiteSourceSettings = whiteSourceSettings;
+    }
+
+    @Override
+    public WhiteSourceCollector getCollector() {
+        return WhiteSourceCollector.prototype(whiteSourceSettings.getServers());
+    }
+
+    @Override
+    public BaseCollectorRepository<WhiteSourceCollector> getCollectorRepository() {
+        return whiteSourceCollectorRepository;
+    }
+
+    @Override
+    public String getCron() {
+        return whiteSourceSettings.getCron();
+    }
+
+    public long getWaitTime() {
+        return whiteSourceSettings.getWaitTime();
+    }
+
+
+    public int getRequestRateLimit() {
+        return whiteSourceSettings.getRequestRateLimit();
+    }
+
+    public long getRequestRateLimitTimeWindow() {
+        return whiteSourceSettings.getRequestRateLimitTimeWindow();
+    }
+
+
+    @Override
+    public void collect(WhiteSourceCollector collector) {
+        long start = System.currentTimeMillis();
+        collector.getWhiteSourceServers().forEach(instanceUrl -> {
+            logBanner(instanceUrl);
+            whiteSourceSettings.getOrgTokens().forEach(orgToken -> {
+                String orgName = whiteSourceClient.getOrgDetails(instanceUrl, orgToken);
+                List<WhiteSourceProduct> products = whiteSourceClient.getProducts(instanceUrl, orgToken);
+                List<WhiteSourceComponent> projects = new ArrayList<>();
+                for (WhiteSourceProduct product : products) {
+                    projects.addAll(whiteSourceClient.getAllProjectsForProduct(instanceUrl, product, orgToken, orgName));
+                }
+                log("Fetched projects for Organization : " + orgName + ":  total projects :" + projects.size(), start);
+                List<WhiteSourceComponent> existingComponents = whiteSourceComponentRepository.findByCollectorIdIn(Stream.of(collector.getId()).collect(Collectors.toList()));
+                addNewApplications(projects, existingComponents, collector);
+                refreshData(enabledApplications(collector), getRequestRateLimit(),
+                        getRequestRateLimitTimeWindow(), getWaitTime(), instanceUrl);
```

Comment:
Line 96 : Yes, it should be call at instanceUrl, but I thought it will be an unnecessary call if org is invalid. So moved it to org loop and called it after getting org details.
Line 97 and 98, there are at org level.